remix-simulator/src/methods/filters.js

```diff
@@ -29,4 +28,36 @@ Filters.prototype.eth_unsubscribe = function (payload, cb) {
   cb(null, true)
 }
 
+Filters.prototype.eth_newFilter = function (payload, cb) {
```

Comment:
Any reason to use a Callback design instead of a promise one ?

Reply:
it works well with the current design of the remix-libs, promises would (currently) unnecessarily complicate things at this point